# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: 3daeacd08336
# Full Hash: 3daeacd0833682a474d9bd1adc8817474a7602be
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 6: Move NotificationObserver functions to NotificationUtils r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227691
# ==============================================================================

diff -r a497ddb030f6 -r 3daeacd08336 dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Thu Nov 07 17:32:05 2024 +0000
+++ b/dom/notification/NotificationParent.cpp	Thu Nov 07 17:32:05 2024 +0000
@@ -19,6 +19,20 @@
 NS_IMETHODIMP
 NotificationParent::Observe(nsISupports* aSubject, const char* aTopic,
                             const char16_t* aData) {
+  if (!strcmp("alertdisablecallback", aTopic)) {
+    return RemovePermission(mPrincipal);
+  }
+  if (!strcmp("alertsettingscallback", aTopic)) {
+    return OpenSettings(mPrincipal);
+  }
+  if (!strcmp("alertshow", aTopic)) {
+    (void)NS_WARN_IF(NS_FAILED(
+        AdjustPushQuota(mPrincipal, NotificationStatusChange::Shown)));
+  }
+  if (!strcmp("alertfinished", aTopic)) {
+    (void)NS_WARN_IF(NS_FAILED(
+        AdjustPushQuota(mPrincipal, NotificationStatusChange::Closed)));
+  }
   return NS_OK;
 }
 